dashboard/app/models/school_stats_by_year.rb

```diff
@@ -93,4 +93,8 @@ def self.merge_from_csv(filename, options = {col_sep: "\t", headers: true, quote
   def has_high_school_grades?
     grade_09_offered || grade_10_offered || grade_11_offered || grade_12_offered || grade_13_offered
   end
+
+  def has_k8_grades?
+    grade_kg_offered || grade_01_offered || grade_02_offered || grade_03_offered || grade_04_offered || grade_05_offered || grade_06_offered || grade_07_offered || grade_08_offered
```

Comment:
I feel like Ruby must have some magic to generate this list programatically :)

Reply:
You could do something like this:
```ruby
irb(main):014:0> (['kg'] + (9..13).to_a).map{|n| "grade_#{n.to_s.rjust(2,'0')}_offered"}
=> ["grade_kg_offered", "grade_09_offered", "grade_10_offered", "grade_11_offered", "grade_12_offered", "grade_13_offered"]
```

Then you'll need to run those strings through `read_attribute` and `reduce` or something similar to `||` them together. I'm not sure that will be more readable.

Reply:
Not that I could determine.